src/unix/mod.rs

```diff
@@ -143,6 +143,9 @@ cfg_if! {
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]
         extern {}
+    } else if #[cfg(all(target_vendor = "rumprun", target_os = "netbsd"))] {
+        #[link(name = "m")]
```

Comment:
Rebased and added a comment to clarify why we omit `-lc`. Note that while I only observed the assertion failure when linking Rust, I don't think the crash is Rust's fault. It probably is caused by Rumpruns use of  `ld -r` which has triggered bugs in binutils before, see rumpkernel/rumprun-packages#19